insights/src/main/java/org/jboss/as/telemetry/extension/InsightsRequestHandler.java

```diff
@@ -0,0 +1,76 @@
+package org.jboss.as.insights.extension;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.SimpleOperationDefinition;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.dmr.ModelNode;
+
+public class InsightsRequestHandler implements OperationStepHandler {
+
+    private static final String OPERATION_NAME = "enable";
+
+    static final InsightsRequestHandler INSTANCE = new InsightsRequestHandler();
+
+    static final SimpleOperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(OPERATION_NAME, InsightsExtension.getResourceDescriptionResolver(null))
+            .setRuntimeOnly()
+            .addParameter(InsightsSubsystemDefinition.RHNUID)
+            .addParameter(InsightsSubsystemDefinition.RHNPW)
+            .addParameter(InsightsSubsystemDefinition.PROXYURL)
+            .addParameter(InsightsSubsystemDefinition.PROXYPORT)
+            .addParameter(InsightsSubsystemDefinition.PROXYPASSWORD)
+            .addParameter(InsightsSubsystemDefinition.PROXYUSER)
+            .build();
+
+    private InsightsRequestHandler() {
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+
+        final String rhnUid = operation.require(InsightsExtension.RHNUID).asString();
+        final String rhnPw = operation.require(InsightsExtension.RHNPW).asString();
+
+        if(operation.has(InsightsExtension.PROXY_PASSWORD) && operation.has(InsightsExtension.PROXY_USER)) {
+            String proxyPwd = operation.get(InsightsExtension.PROXY_PASSWORD).asString();
```

Comment:
this should be InsightsSubsystemDefinition.RHNPW.resolveModelAttribute(operation, context)
and similarly all others.